core/watcher/watchertest/watcher.go

```diff
@@ -138,7 +138,18 @@ func (w *WatcherC[T]) Check(assertion WatcherAssert[T]) {
 		select {
 		case actual, ok := <-w.Watcher.Changes():
 			w.c.Logf("WatcherC Watcher.Changes() => %# v", actual)
-			w.c.Assert(ok, jc.IsTrue)
+			if !ok {
+				wait := make(chan error)
+				go func() {
+					wait <- w.Watcher.Wait()
+				}()
+				select {
+				case <-time.After(testing.LongWait):
+					w.c.Fatalf("watcher never stopped")
+				case err := <-wait:
```

Comment:
Why not read `w.Watcher.Wait()` directly instead of the assignment and Goroutine?

Reply:
Agree with @manadart here. I missed this when reviewing

Reply:
Answered here https://github.com/juju/juju/pull/19595#discussion_r2057796909